File: solutions/javascript/space-age/1/space-age.js
//
// This is only a SKELETON file for the 'Space Age' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const age = (planet, ageInSeconds) => {
  const EARTH_YEAR_IN_SECONDS = 31_557_600;

  const orbitalPeriods = {
    mercury: 0.2408467,
    venus: 0.61519726,
    earth: 1.0,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
  };

  if (!(planet in orbitalPeriods)) {
    throw new Error('not a planet');
  }
  
  const earthAge = ageInSeconds / EARTH_YEAR_IN_SECONDS;

  const planetAge = earthAge / orbitalPeriods[planet];

  return Math.round(planetAge * 100) / 100;
};

File: solutions/python/space-age/1/space_age.py
class SpaceAge:
    def __init__(self, seconds):
        self.seconds = seconds
        self.earth_year_seconds = 31557600  # seconds in one Earth year
        
        # Orbital periods in Earth years
        self.orbital_periods = {
            'mercury': 0.2408467,
            'venus': 0.61519726,
            'earth': 1.0,
            'mars': 1.8808158,
            'jupiter': 11.862615,
            'saturn': 29.447498,
            'uranus': 84.016846,
            'neptune': 164.79132
        }
    
    def _calculate_age(self, orbital_period):
        """Helper method to calculate age on a planet given its orbital period."""
        earth_years = self.seconds / self.earth_year_seconds
        planet_years = earth_years / orbital_period
        return round(planet_years, 2)
    
    def on_mercury(self):
        return self._calculate_age(self.orbital_periods['mercury'])
    
    def on_venus(self):
        return self._calculate_age(self.orbital_periods['venus'])
    
    def on_earth(self):
        return self._calculate_age(self.orbital_periods['earth'])
    
    def on_mars(self):
        return self._calculate_age(self.orbital_periods['mars'])
    
    def on_jupiter(self):
        return self._calculate_age(self.orbital_periods['jupiter'])
    
    def on_saturn(self):
        return self._calculate_age(self.orbital_periods['saturn'])
    
    def on_uranus(self):
        return self._calculate_age(self.orbital_periods['uranus'])
    
    def on_neptune(self):
        return self._calculate_age(self.orbital_periods['neptune'])
